src/Core/Picking.js

```diff
@@ -137,11 +137,20 @@ export default {
             view.viewToNormalizedCoords(mouse),
             view.camera.camera3D);
         const intersects = raycaster.intersectObject(object, true);
+        function layerIdToLayer(id) { return view.getLayers(l => l.id == id); }
         for (const inter of intersects) {
             inter.layer = findLayerIdInParent(inter.object);
+            const protocol = layerIdToLayer(inter.layer)[0].protocol;
+            switch (protocol) {
```

Comment:
We already have a mechanism for custom picking logic per protocol, this file shouldn't have any protocol specific code.

Reply:
Where will you place it?

Reply:
See #635 (you were the reviewer on this PR). In this precise case, have a `pick3dTilesObjects` that calls this method and add some specific 3dtiles properties on the result can be one possible solution.